plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraTypeManager.java

```diff
@@ -308,11 +316,42 @@ private String buildArrayValueFromSetType(GettableByIndex row, int position, Dat
         return buildArrayValue((Collection<?>) row.getObject(position), setType.getElementType());
     }
 
-    private String buildArrayValueFromListType(GettableByIndex row, int position, DataType type)
+    private Block buildArrayValueFromListType(GettableByIndex row, int position, DataType type)
     {
         checkArgument(type instanceof ListType, "Expected to deal with an instance of %s class, got: %s", ListType.class, type);
         ListType listType = (ListType) type;
-        return buildArrayValue((Collection<?>) row.getObject(position), listType.getElementType());
+        ArrayType trinoArrayType = getTrinoArrayType(listType.getElementType());
+        return buildArrayValueFromCollection((Collection<?>) row.getObject(position), trinoArrayType);
```

Comment:
row.getList method is supported by datastax driver if you like to go that route instead of row.getObject(position). 

Following is what I did.

`
TypeCodec codec = row.codecRegistry().codecFor(trinoArrayType);
        return buildTrinoArrayValueFromCollection(row.getList(position, codec.getJavaType().getRawType()), trinoArrayType);
        `